packages/orchestration/src/exos/chain-hub.js

```diff
@@ -120,22 +122,25 @@ export const makeChainHub = (agoricNames, zone = makeHeapZone()) => {
     /**
      * @template {string} K
      * @param {K} chainName
-     * @returns {Promise<ActualChainInfo<K>>}
+     * @returns {Vow<ActualChainInfo<K>>}
      */
-    async getChainInfo(chainName) {
+    getChainInfo(chainName) {
       // Either from registerChain or memoized remote lookup()
       if (chainInfos.has(chainName)) {
-        // @ts-expect-error cast
-        return chainInfos.get(chainName);
+        return /** @type {Vow<ActualChainInfo<K>>} */ (
+          watch(chainInfos.get(chainName))
+        );
       }
 
-      const chainInfo = await E(agoricNames)
-        .lookup(CHAIN_KEY, chainName)
-        .catch(_cause => {
+      return watch(E(agoricNames).lookup(CHAIN_KEY, chainName), {
+        onFulfilled: chainInfo => {
+          chainInfos.init(chainName, chainInfo);
+          return chainInfo;
+        },
+        onRejected: _cause => {
```

Comment:
I'm (still) uneasy throwing away `_cause`. But I don't have an alternative to suggest. :-/